resources/cloudfront_cache_policies.go

```diff
@@ -0,0 +1,133 @@
+			},
+			{
+				Name:     "cookies_behavior",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.CookiesConfig.CookieBehavior"),
+			},
+			{
+				Name:     "cookies",
+				Type:     schema.TypeStringArray,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.CookiesConfig.Cookies.Items"),
+			},
+			{
+				Name:     "enable_accept_encoding_gzip",
+				Type:     schema.TypeBool,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.EnableAcceptEncodingGzip"),
+			},
+			{
+				Name:     "headers_behavior",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.HeadersConfig.HeaderBehavior"),
+			},
+			{
+				Name:     "headers",
+				Type:     schema.TypeStringArray,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.HeadersConfig.Headers.Items"),
+			},
+			{
+				Name:     "query_strings_behavior",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.QueryStringsConfig.QueryStringBehavior"),
+			},
+			{
+				Name:     "query_strings",
+				Type:     schema.TypeStringArray,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.QueryStringsConfig.QueryStrings.Items"),
+			},
+			{
+				Name:     "enable_accept_encoding_brotli",
+				Type:     schema.TypeBool,
+				Resolver: schema.PathResolver("CachePolicy.CachePolicyConfig.ParametersInCacheKeyAndForwardedToOrigin.EnableAcceptEncodingBrotli"),
+			},
+			{
+				Name:     "resource_id",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("CachePolicy.Id"),
+			},
+			{
+				Name:     "last_modified_time",
+				Type:     schema.TypeTimestamp,
+				Resolver: schema.PathResolver("CachePolicy.LastModifiedTime"),
+			},
+			{
+				Name: "type",
+				Type: schema.TypeString,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchCloudfrontCachePolicies(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	var config cloudfront.ListCachePoliciesInput
+	c := meta.(*client.Client)
+	s := c.Services()
+	svc := s.Cloudfront
+	for {
+		response, err := svc.ListCachePolicies(ctx, nil, func(options *cloudfront.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return err
+		}
+
+		if response.CachePolicyList != nil {
```

Comment:
you can push nil inside to the res channel and it will work as well. (its okay this way as well)

Reply:
access to response.CachePolicyList.Items can cause a runtime error